android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageKeystoreRSAECB.java

```diff
@@ -0,0 +1,322 @@
+            decryptedUsername = decryptBytes(key, username);
+            decryptedPassword = decryptBytes(key, password);
+        } catch (UserNotAuthenticatedException e) {
+            mDecryptParams = new CipherDecryptionParams(decryptionResultHandler, key, username, password);
+            if (!canStartFingerprintAuthentication()) {
+                throw new CryptoFailedException("Could not start fingerprint Authentication");
+            }
+            try {
+                // The Cipher is locked, we will decrypt once fingerprint is recognised.
+                startFingerprintAuthentication();
+            } catch (Exception e1) {
+                e1.printStackTrace();
+                throw new CryptoFailedException("Could not start fingerprint Authentication", e1);
+            }
+            return;
+        }
+
+        // The Cipher is unlocked, we can decrypt straight away.
+        decryptionResultHandler.onDecrypt(new DecryptionResult(decryptedUsername, decryptedPassword, SecurityLevel.ANY), null);
+    }
+
+    private byte[] encryptString(Key key, String service, String value) throws CryptoFailedException {
+        try {
+            Cipher cipher = Cipher.getInstance(ENCRYPTION_TRANSFORMATION);
+            cipher.init(Cipher.ENCRYPT_MODE, key);
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            // encrypt the value using a CipherOutputStream
+            CipherOutputStream cipherOutputStream = new CipherOutputStream(outputStream, cipher);
+            cipherOutputStream.write(value.getBytes("UTF-8"));
+            cipherOutputStream.close();
+            return outputStream.toByteArray();
+        } catch (Exception e) {
+            throw new CryptoFailedException("Could not encrypt value for service " + service, e);
+        }
+    }
+
+    private Cipher getDecryptionCipher(Key key) throws CryptoFailedException, UserNotAuthenticatedException, KeyPermanentlyInvalidatedException {
+        try {
+            Cipher cipher = Cipher.getInstance(ENCRYPTION_TRANSFORMATION);
+            // read the initialization vector from the beginning of the stream
+            cipher.init(Cipher.DECRYPT_MODE, key);
+
+            return cipher;
+        } catch (UserNotAuthenticatedException | KeyPermanentlyInvalidatedException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new CryptoFailedException("Could not generate cipher", e);
+        }
+    }
+
+    private String decryptBytes(Key key, byte[] bytes) throws CryptoFailedException, UserNotAuthenticatedException, KeyPermanentlyInvalidatedException {
+        try {
+            Cipher cipher = getDecryptionCipher(key);
+            ByteArrayInputStream inputStream = new ByteArrayInputStream(bytes);
+            // decrypt the bytes using a CipherInputStream
+            CipherInputStream cipherInputStream = new CipherInputStream(
```

Comment:
this part can be extracted into a utility function and shared in `CipherStorageKeystoreAESCBC.java` and `CipherStorageKeystoreRSAECB.java`